parser/parser.go

```diff
@@ -0,0 +1,393 @@
+package parser
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/alecthomas/participle"
+	"github.com/alecthomas/participle/lexer"
+	"github.com/flynn/biscuit-go"
+	"github.com/flynn/biscuit-go/datalog"
+)
+
+var (
+	ErrVariableInFact = errors.New("parser: a fact cannot contain any variable")
+)
+
+var DefaultParserOptions = []participle.Option{
```

Comment:
Should this be exported? Is there anything that users of the package need to modify?